### ISSUE TITLE
[enhancement] Aliasing undesirable field names, redux

### ISSUE THREAD

**Issue by estaub** [calebmer/postgraphql#385]
This is related to https://github.com/calebmer/postgraphql/issues/34, but (AFAICT) is not addressed at all by https://github.com/calebmer/postgraphql/issues/42 .

Desirable column name conventions are not necessarily desirable GraphQL field name conventions. Human domain knowledge can lead to much simpler and more obvious field names than those required to disambiguate all possible cases.  

In my case, this applies especially to relations.  For example, I suffix foreign key columns with an "_fk".  Also, in my usage it's rare for a table to reference another by more than one foreign key.  So instead of having a field for e.g., a `parent_fk` reference named `personByParentFk`, I'd really like to have GraphQL users be able to just see a field named `parent`.  

Is there any way currently to do field name-mapping?

If not... while I can imagine some way to embed this in the schema, using e.g. a comment formalization, I think adding JS metadata would be more straightforward for most people.

**Comment by troelsh** [comment]
While i do appreciate the problem you face ( i have a similar one ) i think that we have went down a wrong path in the first place putting any kind of meaning into table or column names. In my own case my problem comes from a naming convention from a ORM i created that used the table naming for reflection on the database.

So i have stuff like this a_u_user = A for Admin module, U for user and user which should have been the actual table name.

Columns in that table are then name a_u_id, a_u_name a_u_email ... etc.

While this have had it's advantages it does give problems when using other ORM's or PostGraphQL. I think we simply have to rename our DB.

I have been fiddling about with doing VIEWS of my tables with more normal names and this might actually be a way to go. 

So to sum up, or maybe ask a question. Have we put ourselves into misery by adding meaning to table and column names in our DB schemas.

**Comment by estaub** [comment]
@troelsh I don't see it that way.  In my usage, there are other considerations, e.g., mappings from the data source names to the database names, with competing requirements.  I have no control over the data source names.  Also, I'd expect that there are a lot of cases where people want to add a GraphQL API with a pre-existing database, where a whole schema remake is prohibitive because of a large body of existing code.  PostgresQL feels over-rigid, _possibly_ with a whiff of religiosity to it's "the DB schema is the only source of metadata" creed.

**Comment by estaub** [comment]
Appears to be at least partially addressed by the work in https://github.com/calebmer/postgraphql/pull/338 .  Closed till that's clearer.
